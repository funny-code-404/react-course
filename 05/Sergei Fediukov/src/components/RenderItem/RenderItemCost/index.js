const RenderItemCost = (props) => {
    const { cost } = props
    return (
        <div>
            <p>Costs:</p>
            <ul>
                {cost.Wood && <li>Wood - <span>{cost.Wood}</span></li>}
                {cost.Gold && <li>Gold - <span>{cost.Gold}</span></li>}
                {cost.Food && <li>Food - <span>{cost.Food}</span></li>}
                {cost.info && <li>Cost - <span>{cost.info}</span></li>}
                {cost.Stone && <li>Stone - <span>{cost.Stone}</span></li>}
            </ul>
        </div>
    )
}
export default RenderItemCost