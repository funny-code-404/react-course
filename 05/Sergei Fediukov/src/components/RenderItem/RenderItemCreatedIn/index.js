import { UniqueCreatedIn } from '../../../pages'
import React from 'react'
class RenderItemCreatedIn extends React.Component {
    state = {
        created_in: false,
    }
    render() {
        const created_in = this.props.created_in
        const hendleCreatedIn = () => {
            this.setState(prevState => ({
                ...prevState,
                created_in: !prevState.created_in
            }))
        }
        return (
            <div>
                <p>Created in:</p>
                <ul>
                    <li>
                        <button onClick={hendleCreatedIn}>
                            {
                                !this.state.created_in ?
                                    created_in.split('/')[created_in.split('/').length - 1].replace('_', ' ')
                                    : 'close'
                            }
                        </button>
                        {
                            this.state.created_in &&
                            <div>
                                <UniqueCreatedIn url={created_in} />
                            </div>
                        }
                    </li>
                </ul>
            </div>
        )
    }
}
export default RenderItemCreatedIn

