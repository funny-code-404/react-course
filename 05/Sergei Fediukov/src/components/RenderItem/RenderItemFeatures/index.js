const RenderItemFeatures = (props) => {
    const { features } = props
    return (
        <>
            {
                features.map(x => {
                    for (const key in x) {
                        return (
                            x[key] &&
                            <p key={key}>{key.charAt(0).toUpperCase() + key.substr(1).replace(/\_/g, ' ')}: <span>{x[key]}</span></p>
                        )
                    }
                })
            }
        </>
    )
}
export default RenderItemFeatures