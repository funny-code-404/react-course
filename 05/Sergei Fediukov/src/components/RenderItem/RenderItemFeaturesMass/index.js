const RenderItemFeaturesMass = (props) => {
    const { features_mass } = props
    return (
        features_mass.map(x => {
            for (const key in x) {
                return (
                    x[key] &&
                    <div key={key}>
                        <p >{key.charAt(0).toUpperCase() + key.substr(1).replace(/\_/g, ' ')} </p>
                        <ul>
                            {x[key].map((x) => <li key={x}>{x}</li>)}
                        </ul>
                    </div>
                )
            }
        })
    )
}
export default RenderItemFeaturesMass