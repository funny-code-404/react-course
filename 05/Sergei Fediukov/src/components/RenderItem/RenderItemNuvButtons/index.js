const RenderItemNavButtons = ({ props }) => {
    const handleGoBack = () => {
        props.goBack()
    }
    const handleRedirect = () => {
        props.push('/')
    }
    return (
        <div>
            <button className='go_back' onClick={handleGoBack}>go back</button>
            <button className='redirect' onClick={handleRedirect}>go home</button>
        </div>
    )

}
export default RenderItemNavButtons 