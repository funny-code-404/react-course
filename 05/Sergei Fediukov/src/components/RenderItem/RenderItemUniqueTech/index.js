import { UniqueTech } from '../../../pages'
import React from 'react'
class RenderItemUniqueTech extends React.Component {
    state = {
        unique_tech: {
            0: false,
            1: false
        },
    }
    render() {
        const unique_tech = this.props.unique_tech
        const hendleUniqueTech = (e) => {
            const i = e.target.attributes[0].value
            this.setState(prevState => ({
                ...prevState,
                unique_tech: {
                    ...prevState.unique_tech,
                    [i]: !prevState.unique_tech[i]
                }
            }))
        }
        return (
            <div>
                <p>Unique tech:</p>
                <ul>
                    {unique_tech.map((x, i) =>
                        <li key={x}>
                            <div>
                                <button onClick={hendleUniqueTech} i={i}>
                                    {
                                        !this.state.unique_tech[i] ?
                                            x.split('/')[x.split('/').length - 1].replace('_', ' ')
                                            : 'close'
                                    }
                                </button>
                            </div>
                            {
                                this.state.unique_tech[i] &&
                                <div>
                                    <UniqueTech url={x} />
                                </div>
                            }
                        </li>)}
                </ul>
            </div>
        )
    }
}
export default RenderItemUniqueTech

