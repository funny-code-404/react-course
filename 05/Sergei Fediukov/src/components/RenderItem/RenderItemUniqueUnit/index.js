import { UniqueUnit } from '../../../pages'
import React from 'react'
class RenderItemUniqueUnit extends React.Component {
    state = {
        unique_unit: {
            0: false,
            1: false
        }
    }
    render() {
        const unique_unit = this.props.unique_unit
        const hendleUniqueUnit = (e) => {
            const i = e.target.attributes[0].value
            this.setState(prevState => ({
                ...prevState,
                unique_unit: {
                    [i]: !prevState.unique_unit[i]
                }
            }))
        }
        return (
            <div className='result_inner' >
                <p>Unique unit:</p>
                <ul>
                    {unique_unit.map((x, i) =>
                        <li key={x}>
                            <div>
                                <button onClick={hendleUniqueUnit} i={i}>
                                    {
                                        !this.state.unique_unit[i] ?
                                            x.split('/')[x.split('/').length - 1].replace('_', ' ')
                                            : 'close'
                                    }
                                </button>
                            </div>
                            {
                                this.state.unique_unit[i] &&
                                <div>
                                    <UniqueUnit url={x} />
                                </div>
                            }
                        </li>)}
                </ul>
            </div>
        )
    }
}
export default RenderItemUniqueUnit

