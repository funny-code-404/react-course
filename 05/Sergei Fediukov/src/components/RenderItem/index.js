import React from 'react'
import SmallLoader from '../Loader/SmallLoader'
import RenderItemNavButtons from './RenderItemNuvButtons'
import RenderItemUniqueUnit from './RenderItemUniqueUnit'
import RenderItemUniqueTech from './RenderItemUniqueTech'
import RenderItemCreatedIn from './RenderItemCreatedIn'
import RenderItemFeatures from './RenderItemFeatures'
import RenderItemFeaturesMass from './RenderItemFeaturesMass'
import RenderItemCost from './RenderItemCost'

class RenderItem extends React.Component {
    state = {
        flag: false,
        applies_to: false,
    }
    componentDidUpdate(prevProps) {
        if ((prevProps.data !== this.props.data)) {
            this.setState({
                flag: true
            })
        }
    }
    render() {
        const data = this.props.data
        const { name, army_type, expansion, team_bonus, civilization_bonus, unique_unit, unique_tech, description, age, created_in, cost, build_time, reload_time, attack_delay, movement_rate, line_of_sight, hit_points, range, attack, armor, accuracy, special, attack_bonus, search_radius } = data
        const features = [{ army_type }, { team_bonus }, { description }, { expansion }, { age }, { build_time }, { reload_time }, { attack_delay }, { movement_rate }, { line_of_sight }, { hit_points }, { range }, { attack }, { armor }, { search_radius }, { accuracy }]
        const features_mass = [{ attack_bonus, special, civilization_bonus }]
        return (<>
            {
                !this.state.flag ?
                    <SmallLoader /> :
                    <div className='result'>
                        {
                            Boolean(this.props.unique) ||
                            <RenderItemNavButtons props={this.props.history} />
                        }
                        <h1>
                            {name}
                        </h1>

                        <RenderItemFeatures features={features} />

                        <RenderItemFeaturesMass features_mass={features_mass} />
                        {
                            cost && <RenderItemCost cost={cost} />
                        }
                        {
                            unique_unit && Boolean(unique_unit.length) &&
                            <RenderItemUniqueUnit unique_unit={unique_unit} />
                        }
                        {
                            unique_tech && Boolean(unique_tech.length) &&
                            <RenderItemUniqueTech unique_tech={unique_tech} />
                        }
                        {
                            created_in &&
                            <RenderItemCreatedIn created_in={created_in} />
                        }
                    </div>
            }
        </>)
    }
}
export default RenderItem