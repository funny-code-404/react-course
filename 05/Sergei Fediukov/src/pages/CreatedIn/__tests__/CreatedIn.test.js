import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import { UniqueCreatedIn } from '..'
import RenderItem from '../../../components/RenderItem';

configure({ adapter: new Adapter() });

describe('Should render correctly "CreatedIn"', () => {
    const mockFn = jest.fn();

    const props = {
        uniqueCreatedIn: { data: '' },
        getUniqueCreatedInDetails: mockFn,
        history: { push: mockFn },
        url: 'test url',
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<UniqueCreatedIn {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('prop-function should be called in componentDidMount', () => {
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
    it('component has children component "RenderItem"', () => {
        expect(wrapper.find(RenderItem)).toHaveLength(1)
    })

})