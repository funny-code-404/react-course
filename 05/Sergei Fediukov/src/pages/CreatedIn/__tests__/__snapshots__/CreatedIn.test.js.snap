// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "CreatedIn" to Match Snapshot 1`] = `
<div>
  <RenderItem
    data=""
    unique="true"
  />
</div>
`;
